### PR TITLE
Re-structure the README file

### PR DESCRIPTION
**Pull request by olksdr**
Moved the Usage section to the top of the file, to make clear how the
provider need to be configured and used. Added config block for TF
required providers and versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,3 +84,34 @@ In order to run the full suite of Acceptance tests, run `make testacc`.
 ```sh
 $ make testacc
 ```
+
+
+### Requirements
+
+-	[Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
```
Comment (Crazybus) [inline README.md]:
```suggestion
- [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
```

File: README.md [hunk 2]
```diff
@@ -96,3 +84,34 @@ In order to run the full suite of Acceptance tests, run `make testacc`.
...
+-	[Go](https://golang.org/doc/install) >= 1.17
```
Comment (Crazybus) [inline README.md]:
```suggestion
- [Go](https://golang.org/doc/install) >= 1.17
```

File: README.md [hunk 3]
```diff
@@ -39,6 +12,21 @@ It is recommended to setup at least minimum security, https://www.elastic.co/gui
 in order to interract with the Elasticsearch and be able to use the provider's full capabilities
```
Comment (gtback) [inline README.md:12]:
```suggestion
in order to interact with the Elasticsearch and be able to use the provider's full capabilities.
```
not part of this PR, but I noticed while reviewing the README.

Comment (olksdr) [reply README.md:12]: Thanks! I will address this in the followup PR
